crates/sncast/src/response/structs.rs

```diff
@@ -151,3 +154,49 @@ pub struct VerifyResponse {
 }
 
 impl CommandResponse for VerifyResponse {}
+
+impl OutputLink for InvokeResponse {
+    const TITLE: &'static str = "invocation";
+
+    fn format_links(&self, service: &str) -> String {
+        format!("transaction: {service}/{:x}", self.transaction_hash.0)
+    }
+}
+
+impl OutputLink for DeployResponse {
+    const TITLE: &'static str = "deployment";
+
+    fn format_links(&self, service: &str) -> String {
+        formatdoc!(
+            "
+            contract: {service}/{:x}
+            transaction: {service}/{:x}
+            ",
+            self.contract_address.0,
+            self.transaction_hash.0
+        )
+    }
+}
+
+impl OutputLink for DeclareResponse {
+    const TITLE: &'static str = "declaration";
+
+    fn format_links(&self, service: &str) -> String {
+        formatdoc!(
+            "
+            class: {service}/{:x}
+            transaction: {service}/{:x}
```

Comment:
Not sure if it's going to be that simple. In case of Starkscan we have an endpoint `/search`, that simply redirects to the class/contract/etc. we're looking for. In case of e.g. Voyager, there is no such endpoint (or at least I haven't found it). This implementation implies that every explorer would have such an endpoint, which is not that obvious.

Reply:
Indeed - Voyager has separate endpoints for block, transactions etc.
What we can do is to narrow available block explorer services to have a control over URL formatting.
I propose introducing an enum with supported services.